File: mvp/sdk/python/saas_ia/api/content_studio.py
"""Content Studio API — multi-format content generation (10 formats)."""

from __future__ import annotations

from typing import Any

from saas_ia.api.base import BaseAPI


class ContentStudioAPI(BaseAPI):
    """Wraps ``/api/content-studio`` endpoints."""

    # -- Projects -----------------------------------------------------------

    async def create_project(
        self,
        title: str,
        *,
        description: str | None = None,
        source_text: str | None = None,
    ) -> dict[str, Any]:
        """Create a content project."""
        body: dict[str, Any] = {"title": title}
        if description:
            body["description"] = description
        if source_text:
            body["source_text"] = source_text
        return await self._post("/api/content-studio/projects", json=body)

    async def list_projects(
        self,
        *,
        limit: int | None = None,
        offset: int | None = None,
    ) -> list[dict[str, Any]]:
        """List content projects."""
        return await self._get(
            "/api/content-studio/projects",
            params={"limit": limit, "offset": offset},
        )

    async def delete_project(self, project_id: str) -> None:
        """Delete a project and all its content pieces."""
        await self._delete(f"/api/content-studio/projects/{project_id}")

    # -- Content generation -------------------------------------------------

    async def generate(
        self,
        project_id: str,
        formats: list[str],
        *,
        tone: str | None = None,
        language: str | None = None,
    ) -> list[dict[str, Any]]:
        """Generate content in one or more formats for a project."""
        body: dict[str, Any] = {"formats": formats}
        if tone:
            body["tone"] = tone
        if language:
            body["language"] = language
        return await self._post(
            f"/api/content-studio/projects/{project_id}/generate",
            json=body,
        )

    async def list_contents(self, project_id: str) -> list[dict[str, Any]]:
        """Get all content pieces for a project."""
        return await self._get(
            f"/api/content-studio/projects/{project_id}/contents"
        )

    async def update_content(
        self,
        content_id: str,
        *,
        content: str | None = None,
        metadata: dict[str, Any] | None = None,
    ) -> dict[str, Any]:
        """Update a content piece."""
        body: dict[str, Any] = {}
        if content is not None:
            body["content"] = content
        if metadata is not None:
            body["metadata"] = metadata
        return await self._put(
            f"/api/content-studio/contents/{content_id}",
            json=body,
        )

    async def regenerate_content(self, content_id: str) -> dict[str, Any]:
        """Regenerate a content piece."""
        return await self._post(
            f"/api/content-studio/contents/{content_id}/regenerate"
        )

    # -- URL-based generation -----------------------------------------------

    async def from_url(
        self,
        url: str,
        formats: list[str],
        *,
        tone: str | None = None,
    ) -> list[dict[str, Any]]:
        """Crawl a URL and generate content from its text."""
        body: dict[str, Any] = {"url": url, "formats": formats}
        if tone:
            body["tone"] = tone
        return await self._post("/api/content-studio/from-url", json=body)

    # -- Reference ----------------------------------------------------------

    async def list_formats(self) -> list[dict[str, Any]]:
        """List available content formats."""
        return await self._get("/api/content-studio/formats")

File: mvp/sdk/typescript/src/api/content-studio.ts
import { BaseAPI } from "./base";
import {
  ContentProject,
  ContentPiece,
  ContentProjectCreateRequest,
  ContentGenerateRequest,
  ContentFromUrlRequest,
  ContentFormat,
  PaginationParams,
} from "../types";

/**
 * Content Studio API — multi-format content generation (10 formats)
 * from topics, source text, or crawled URLs.
 */
export class ContentStudioAPI extends BaseAPI {
  // -- Projects ------------------------------------------------------------

  /** Create a content project. */
  async createProject(data: ContentProjectCreateRequest): Promise<ContentProject> {
    return this._post<ContentProject>("/api/content-studio/projects", data);
  }

  /** List content projects. */
  async listProjects(params?: PaginationParams): Promise<ContentProject[]> {
    return this._get<ContentProject[]>("/api/content-studio/projects", {
      limit: params?.limit,
      offset: params?.offset,
    });
  }

  /** Delete a project and all its content pieces. */
  async deleteProject(id: string): Promise<void> {
    return this._delete(`/api/content-studio/projects/${id}`);
  }

  // -- Content generation --------------------------------------------------

  /** Generate content in one or more formats for a project. */
  async generate(projectId: string, data: ContentGenerateRequest): Promise<ContentPiece[]> {
    return this._post<ContentPiece[]>(
      `/api/content-studio/projects/${projectId}/generate`,
      data
    );
  }

  /** Get all content pieces for a project. */
  async listContents(projectId: string): Promise<ContentPiece[]> {
    return this._get<ContentPiece[]>(
      `/api/content-studio/projects/${projectId}/contents`
    );
  }

  /** Update a content piece. */
  async updateContent(id: string, data: { content?: string; metadata?: Record<string, unknown> }): Promise<ContentPiece> {
    return this._put<ContentPiece>(`/api/content-studio/contents/${id}`, data);
  }

  /** Regenerate a content piece. */
  async regenerateContent(id: string): Promise<ContentPiece> {
    return this._post<ContentPiece>(`/api/content-studio/contents/${id}/regenerate`);
  }

  // -- URL-based generation ------------------------------------------------

  /** Crawl a URL and generate content from its text. */
  async fromUrl(data: ContentFromUrlRequest): Promise<ContentPiece[]> {
    return this._post<ContentPiece[]>("/api/content-studio/from-url", data);
  }

  // -- Reference data ------------------------------------------------------

  /** List available content formats. */
  async listFormats(): Promise<Array<{ id: ContentFormat; name: string; description: string }>> {
    return this._get("/api/content-studio/formats");
  }
}
